enhancements/etcd/etcd-size-tuning.md

```diff
@@ -0,0 +1,192 @@
+
+### API Extensions
+
+In operator/v1/0000_12_etcd-operator_01_config-TechPreviewNoUpgrade.crd.yaml:
+- Add BackendSizeQuota of type *resource.Quantity
+  - With default '8GiB'.
+  - With validation disallow a decrease in value.
+
+### Implementation Details/Notes/Constraints [optional]
+
+None
+
+### Risks and Mitigations
+
+* We will need to test the larger size to ensure that everything works as expected; especially defragmentation.
+* Defragmentation, and its effects, will need to be tested as the API server will be unresponsive during this time; we will need to understand the general magnitude of defrag time and documentation those expectations.
+* Downgrades may be impacted if:
+  1. The downgrade is to a version without this feature.
+  2. The quota has been set higher than the default.
+  3. The etcd backends are using more space than the default.
+  A mitigation would be to keep the quota at the higher value (with no way of decreasing it), so that the cluster will continue to function and no data will be lost.
+
+### Drawbacks
+
+* There will be a required etcd rollout when changing values, this happens automatically by the Cluster Etcd Operator to re-render the etcd podspec.
+
+## Design Details
+
+### Open Questions [optional]
+
+For each of these, we would need to compare against the current default of 8GiB. Increasing the quota does not necessarily increase the following, but at the extreme case (e.g. increase to 16GiB and using nearly that much) we expect the following to perform differently.
+* How much memory do we require for a given database size?
+* How long does it take to sync a snapshot of a given size to another peer; for reboot/recovery?
+* How long does it take to compact & defrag a database of a given size?
+
+### Test Plan
+
+**Note:** *Section not required until targeted at a release.*
+
+We will need to test a cluster with a larger quota to ensure everything works as expected.
+We should also load the database with a larger quota and make note of the memory requirements of a defragmentation.
+  - We should do this with some values we could assume a customer would use; 16GiB, 32GiB, 100GiB (in the extreme).
+We need to test that it is not possible to shrink the quota once it has be set to a larger value.
+
+### Graduation Criteria
+
+**Note:** *Section not required until targeted at a release.*
+
+None
+
+#### Dev Preview -> Tech Preview
+
+N/A
+
+#### Tech Preview -> GA
+
+- Customer feedback
+- API extension
+- Value change with minimal disruption
+- Performance testing for some larger-than-default values
+- End user documentation
+  - Description of the feature and expectations of memory usages for some largers values.
+  - Steps to change the value.
+
+#### Removing a deprecated feature
+
+N/A
+
+### Upgrade / Downgrade Strategy
+
+N/A
+
+### Version Skew Strategy
+
+N/A
+
+### Operational Aspects of API Extensions
+
+N/A
+
+#### Failure Modes
```

Comment:
Should this resize be preventing during a CPMS scaling event?

Reply:
not sure how setting an env variable via static pods would create any downtime or should be prevented for CPMS updates, care to elaborate?

Reply:
assuming there is timeouts on the resize operation, if attempting to perform this during a control plane scaling event may cause it to fail? Just wondering about what potential failure modes may become apparent.

Reply:
Thank you for the education